### PR TITLE
Adding deployment and fixing API updates

### PR DESCRIPTION
**Pull request by brahmaroutu**
update central controller to pull release tools
Fix issues with api updates
added deployment

### CODE REVIEW

File: samples/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cosi-controller-manager-role
+rules:
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["bucketrequests", "bucketaccessrequests"]
+    verbs: ["get", "list", "watch", "create", "delete"]
```
Comment (jeffvance) [inline samples/rbac.yaml]: why is "BR/BAR create" needed for this ClusterRole?

File: samples/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["buckets", "bucketaccess"]
+    verbs: ["get", "list", "watch", "update", "create", "delete"]
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["buckets","bucketrequests", "bucketclass","bucketaccess","bucketaccessrequests", "bucketaccessclass"]
+    verbs: ["get", "list", "watch"]
```
Comment (jeffvance) [inline samples/rbac.yaml]: Does the central controller "watch" BCs and BACs?

File: samples/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  # replace with non-default namespace name
+  namespace: default
+  name: cosi-controller-manager
+rules:
+# Only one of the following rules for endpoints or leases is required based on
+# what is set for `--leader-election-type`. Endpoints are deprecated in favor of Leases.
+- apiGroups: [""]
+  resources: ["endpoints"]
```
Comment (jeffvance) [inline samples/rbac.yaml]: Probably don't understand the comment above, but if endpoints are deprecated in favor of leases, and you include leases below, why do we need enpoints here?

Comment (brahmaroutu) [reply samples/rbac.yaml]: Fixed

File: samples/rbac.yaml (hunk 2 above)
Comment (brahmaroutu) [reply samples/rbac.yaml]: Fixed

File: samples/rbac.yaml (hunk 1 above)
Comment (brahmaroutu) [reply samples/rbac.yaml]: Fixed, I did not pay close attention and gave more permission than required. Thanks for details review on this @jeffvance

**Pull request closed without merging**